SmartSyncExplorerReactNative/js/ContactScreen.js

```diff
@@ -61,6 +61,34 @@ const ContactScreen = createReactClass({
         storeMgr.saveContact(contact, () => {this.props.navigator.pop()});
     },
 
+    renderErrorIfAny() {
+        var errorMessage = null;
+        const lastError = this.state.contact.__last_error__;
+        if (lastError) {
+            try {
+                errorMessage = new RegExp('.*message = "([^"]*)".*', 'm').exec(lastError)[1];
```

Comment:
This is not pretty.

However I just realized, I passed the raw response on [Android](https://github.com/forcedotcom/SalesforceMobileSDK-Android/blob/master/libs/SmartSync/src/com/salesforce/androidsdk/smartsync/target/SyncUpTarget.java#L204) but not on [iOS](https://github.com/forcedotcom/SalesforceMobileSDK-iOS/blob/master/libs/SmartSync/SmartSync/Classes/Target/SFSyncUpTarget.m#L229). If I were to pass the raw response, we could do a `JSON.parse(lastError).message`.